sentry-sidekiq/lib/sentry/sidekiq/sentry_context_middleware.rb

```diff
@@ -63,13 +74,23 @@ def finish_transaction(transaction, status)
     end
 
     class SentryContextClientMiddleware
-      def call(_worker_class, job, _queue, _redis_pool)
+      def call(worker_class, job, queue, _redis_pool)
         return yield unless Sentry.initialized?
 
         user = Sentry.get_current_scope.user
         job["sentry_user"] = user unless user.empty?
         job["trace_propagation_headers"] ||= Sentry.get_trace_propagation_headers
-        yield
+
+        Sentry.with_child_span(op: "queue.publish", description: "Enqueue #{worker_class}") do |span|
+          # Set span data
+          if span
+            span.set_data("messaging.message.id", job["jid"])
+            span.set_data("messaging.destination.name", queue)
+            span.set_data("messaging.message.body.size", job.to_json.size)
```

Comment:
I wonder if there's a better way to do this. Dumping the job to json (an extra time) on every enqueue sounds not cheap.

Reply:
I was wondering as well - I couldn't really figure out a good way, to give a good insight to size. Could be `job.args.size` but thats misleading; counting only the amount of arguments, not their actual size (Assuming the remaining payload is somewhat constant in size (retry, queue, class, created_at and enqueued_at keys).

Reply:
Having the accurate size of what's being pushed to Redis is cool, but IMO not worth the cost of encoding the entire thing twice.

Reply:
I'm inclined to agree. I'll remove this one - I'm thinking core sentry team can make the final call.